src/api/cvc4cpp.cpp

```diff
@@ -4583,29 +4654,28 @@ void Solver::push(uint32_t nscopes) const
 /**
  *  ( reset-assertions )
  */
-void Solver::resetAssertions(void) const { d_smtEngine->resetAssertions(); }
+void Solver::resetAssertions(void) const
+{
+  CVC4_API_SOLVER_TRY_CATCH_BEGIN;
+  d_smtEngine->resetAssertions();
+  CVC4_API_SOLVER_TRY_CATCH_END;
+}
 
 // TODO: issue #2781
 void Solver::setLogicHelper(const std::string& logic) const
 {
   CVC4_API_CHECK(!d_smtEngine->isFullyInited())
       << "Invalid call to 'setLogic', solver is already fully initialized";
-  try
-  {
-    CVC4::LogicInfo logic_info(logic);
-    d_smtEngine->setLogic(logic_info);
-  }
-  catch (CVC4::IllegalArgumentException& e)
-  {
-    throw CVC4ApiException(e.getMessage());
-  }
+  CVC4::LogicInfo logic_info(logic);
+  d_smtEngine->setLogic(logic_info);
```

Comment:
Don't we need `CVC4_API_SOLVER_TRY_CATCH_BEGIN` and `CVC4_API_SOLVER_TRY_CATCH_END` in this method?

Reply:
No, not in the helper. It's guarded in the function that calls it.

Reply:
I removed the helper for now, it was only used once.

Reply:
Ah I missed the fact that this was a helper, my bad (I just saw that we previously were catching the exception).